datalad_registry/dataset_urls.py

```diff
@@ -36,17 +36,26 @@ def url(ds_id: str, url_encoded: str) -> Any:
     except InvalidURL:
         return jsonify(message="Invalid encoded URL"), 400
 
-    result = db.session.query(URL).filter_by(url=url, ds_id=ds_id)
+    result = db.session.query(URL).filter_by(url=url)
     # even here could lead to sqlite locking issue, I guess if
     # collect_dataset_info is in progress:
     # https://github.com/datalad/datalad-registry/issues/34
     row_known = result.first()
 
+    # We can't replace this check by adding `ds_id=ds_id` to the filter query,
+    # as then entries without ds_id set would not be returned, as though they
+    # were unregistered, leading to PATCH requests trying to register the URLs
+    # again by adding a row to the database, causing an error.
+    if row_known is not None and row_known.ds_id != ds_id:
+        return jsonify(message="UUID does not match value registered for URL"), 400
```

Comment:
I think it is ok for now, but let's add a comment to figure out for a better way to handle such use cases here:
```suggestion
        # TODO: figure out/provide logic/responses for datasets/urls which get their .id changed,
        # or yet to be processed (i.e. `not row_known.processed`)
        return jsonify(message="UUID does not match value registered for URL"), 400
```
?

Reply:
Comment added.